### PR TITLE
refactor: port amount.h

### PR DESCRIPTION
**Pull request by div72**

### CODE REVIEW

**Comment by div72** [comment]
Randomly getting stuck on testnet with `tx index not found`, will investigate further.

File: src/gridcoin/staking/difficulty.cpp [hunk 1]
```diff
@@ -373,7 +374,7 @@ double GRC::GetEstimatedTimetoStake(bool ignore_staking_status, double dDiff, do
     int64_t nTimePrev = nCurrentTime;
     int64_t nDeltaTime = 0;
     int64_t nThrows = 0;
-    int64_t nCoinWeight = 0;
+    CAmount nCoinWeight = 0;
```
Comment (cyrossignol) [inline src/gridcoin/staking/difficulty.cpp]: Not technically a money value.

File: src/gridcoin/staking/kernel.cpp [hunk 2]
```diff
@@ -527,9 +528,9 @@ uint256 GRC::CalculateStakeHashV8(
     return ss.GetHash();
 }
 
-int64_t GRC::CalculateStakeWeightV8(const CTransaction &CoinTx, unsigned CoinTxN)
+CAmount GRC::CalculateStakeWeightV8(const CTransaction &CoinTx, unsigned CoinTxN)
```
Comment (cyrossignol) [inline src/gridcoin/staking/kernel.cpp]: Ditto.

File: src/gridcoin/staking/kernel.cpp [hunk 3]
```diff
@@ -601,7 +602,7 @@ bool GRC::CheckProofOfStakeV8(
     hashProofOfStake = CalculateStakeHashV8(header, txPrev, prevout.n, tx.nTime, StakeModifier);
 
     //Stake refactoring TomasBrod
-    int64_t Weight = CalculateStakeWeightV8(txPrev, prevout.n);
+    CAmount Weight = CalculateStakeWeightV8(txPrev, prevout.n);
```
Comment (cyrossignol) [inline src/gridcoin/staking/kernel.cpp]: Ditto.

File: src/gridcoin/staking/kernel.h [hunk 4]
```diff
@@ -103,5 +104,5 @@ uint256 CalculateStakeHashV8(
     unsigned nTimeTx,
     uint64_t StakeModifier);
 
-int64_t CalculateStakeWeightV8(const CTransaction &CoinTx, unsigned CoinTxN);
+CAmount CalculateStakeWeightV8(const CTransaction &CoinTx, unsigned CoinTxN);
```
Comment (cyrossignol) [inline src/gridcoin/staking/kernel.h]: Ditto.

File: src/util.h [hunk 5]
```diff
@@ -50,7 +50,6 @@
 #include <stdint.h>
 #include <inttypes.h>
 
-static const int64_t COIN = 100000000;
 static const int64_t COIN_PLACES = 8;
```
Comment (cyrossignol) [inline src/util.h]: The `COIN_PLACES` constant isn't used anymore&mdash;let's clean it up.

File: src/util.h [hunk 6]
```diff
@@ -50,7 +50,6 @@
...
 static const int64_t CENT = 1000000;
```
Comment (cyrossignol) [inline src/util.h]: I think it makes sense to move `CENT` to amount.h as well.

**Review by cyrossignol — COMMENTED** [review]
The TX index problem does not seem related to this PR. 

Thanks for porting... I was starting to get nervous when redefining those constants in the headers 🙂 

~Please rebase&mdash;I think there is one more place in gridcoin/contract/contract.h that needs amount.h.~ Nope, you got it!

**Review by cyrossignol — APPROVED** [review]

**Comment by div72** [comment]
> The TX index problem does not seem related to this PR.

Did another sync from zero using the development branch, stuck on the same block so probably an issue on my end not related to this PR.

**Review by jamescowens — APPROVED** [review]

**Pull request merged**